### PR TITLE
Extracting an API for the Virtual Machine

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
Issue: https://github.com/aionnetwork/aion/issues/734
PR: https://github.com/aionnetwork/aion/pull/761

### CODE REVIEW

File: src/org/aion/api/type/core/tx/AionTransaction.java [hunk 1]
```diff
@@ -14,6 +14,11 @@
 import org.aion.crypto.SignatureFac;
 import org.aion.rlp.RLP;
 import org.aion.rlp.RLPList;
+import org.aion.vm.api.interfaces.Address;
+
+// NOTE: a number of methods are now duplicated here (but with different names) because of issue
+// #734.
+// This should be eventually cleaned up, but is not currently a priority.
```
Comment (aion-kelvin) [inline src/org/aion/api/type/core/tx/AionTransaction.java]: Might want to create an issue or jira to track and link it here

**Comment by AlexandraRoatis** [comment]
@aion-kelvin these are the current implementations:
https://github.com/aionnetwork/aion/blob/vm-api-pr/modAionBase/src/org/aion/base/type/AionAddress.java
https://github.com/aionnetwork/vm_api/blob/vm-api-pr/src/org/aion/vm/api/interfaces/Address.java
This is only a first step in the refactoring. The objective would be to have most modules reference only the interface but the interface is not fully specified yet.

**Comment by aion-kelvin** [comment]
Spoke with Jeff and Nick a bit about AionAddress/Address.  Conclusion was probably they don't need to be separated into interface and class, but we can fix that later.

**Comment by AlexandraRoatis** [comment]
will be reopened in stages

**Pull request closed without merging**